### PR TITLE
Rvar matrix stats

### PR DESCRIPTION
**Pull request by mjskay**
#### Summary

Dunno if you want this before the next CRAN submission, but given discussion in #197 I figured I would submit it now.
This PR follows #190 to use functions from matrixStats to speed up rvar summary functions (in some cases, substantially so). It also adds a NEWS entry for this change and for #190, since it did not have one.

There are also some minor changes to base cases for some of the rvar summary functions that I think make them more consistent. Previously, for some summary functions there was a distinction between `rvar_XXX()` and `rvar_XXX(NULL)`; e.g. `rvar_mean()` would raise an error and `rvar_mean(NULL)` would return `NA`. This was to be consistent with *some* base functions that do this (e.g. `mean()` is an error and `mean(numeric())` is `NA`). Yet some do not return an error (e.g. `min()` returns `Inf` with a warning, as does `min(numeric())`). But the semantics of the rvar summary functions is that they take any number of rvars and collapse them into a single rvar vector before computing the summary, so an empty list and a NULL value should be equivalent; thus, I think the semantics of `min()` are better to mimic. Thus the following changes

- `rvar_mean()`, `rvar_median()`, `rvar_sd()`, `rvar_mad()`, and `rvar_var()` now all return `rvar(NA)` when given either an empty argument list or a `NULL` argument, consistent with (e.g.) `mean(numeric())`, `sd(numeric())`, etc. Also consistent with `sd(numeric())` and others, these functions no longer raise warnings in these cases.
- Following `matrixStats::rowMins()`, `::rowMaxs()`, etc., `rvar_min()`, `rvar_max()`, and `rvar_range()` return `rvar(Inf)`, `rvar(-Inf)`, and `as_rvar(c(Inf, -Inf))` *without* a warning. Previously these returned warnings, but I think these base case values are semantically consistent so don't need warnings per se; this is also the approach taken by the matrixStats functions.

I think these changes make the interface to the summary functions more internally consistent.

#### Copyright and Licensing

By submitting this pull request, the copyright holder is agreeing to 
license the submitted work under the following licenses:

- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

File: tests/testthat/test-summarise_draws.R [hunk 1]
```diff
@@ -82,12 +82,12 @@ test_that(paste(
   x <- as_draws_array(test_array)
   sum_x <- summarise_draws(x)
   parsum_x <- summarise_draws(x, .cores = cores)
-  expect_identical(sum_x, parsum_x)
```
Comment (mjskay) [inline tests/testthat/test-summarise_draws.R:85]: I had to change this test and the following one to expect_equal as I was getting "equal but not identical" errors for them on my machine. Not sure why, but I assume this is related to changes to matrixStats impacting parallel execution in some way? In any case I don't think we need to worry if the parallel version produces identical results as long as it is equal down to the default low tolerance?

**Comment by mjskay** [comment]
Also, the codecov decreases here are because: (1) I have left in the `Summary.rvar()` generic in the very unlikely case something calls it in the future (all functions I am aware of that could call it now have faster specific implementations using matrixStats, so there aren't any existing functions that call it) and (2) I have left in `summarise_rvar_within_draws()` even though it is no longer used in favor of `summarise_rvar_within_draws_via_matrix()` as it may be useful in the future for implementing summary functions that do not have matrixStats equivalents.

File: tests/testthat/test-summarise_draws.R (hunk 1 above)
Comment (paul-buerkner) [reply tests/testthat/test-summarise_draws.R:85]: I agree. `expect_identical` is likely too strict in many cases.

Comment (mjskay) [reply tests/testthat/test-summarise_draws.R:85]: Great!

**Comment by paul-buerkner** [comment]
Thank you, the PR looks good to me. Merging this now.

**Pull request merged**